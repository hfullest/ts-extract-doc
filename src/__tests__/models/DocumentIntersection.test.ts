import { resolve } from 'path';
import { extractTsToMarkdown, genrateDocument } from '../..';

describe('DocumentIntersection', () => {
  const sourcePath = resolve(__dirname, '../fixtures/intersection.ts');
  it('文档模型解析:', () => {
    const documents = genrateDocument(sourcePath);
    const result = documents.map(doc=>doc.map(it=>it.toTypeString())?.join('\n')).join('\n\n')
    expect(result).toMatchSnapshot();
  });

  it('快照', () => {
    const result = extractTsToMarkdown(sourcePath);
    expect(result).toMatchSnapshot();
  });
});
