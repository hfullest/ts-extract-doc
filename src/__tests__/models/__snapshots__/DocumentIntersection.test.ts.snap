// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentIntersection 快照 1`] = `
"### UiButtonProps

UiButton 属性


#### 属性

|名称|说明|类型|默认值|版本|
|:--:|:--:|:--:|:---:|:--:|
|a |-|string|-|-|
|b ~(可选)~ |属性b|number|-|-|
|c ||{ c1: string; c2: () => string }|-|-|
|native ~(可选)~ |是否使用原生按钮|N|-|-|
|children ~(可选)~ |-|React.ReactNode|-|-|
|ant ~(可选)~ *~(已废弃)~*|已默认启用\`Antd Button\`，该属性已遗弃，未来版本将移除该属性|never|-|-|
|compat |测试嵌套|boolean|-|-|


### CompatButtonProps

#### 属性

|名称|说明|类型|默认值|版本|
|:--:|:--:|:--:|:---:|:--:|
|native ~(可选)~ |是否使用原生按钮|N|-|-|
|children ~(可选)~ |-|React.ReactNode|-|-|
|ant ~(可选)~ *~(已废弃)~*|已默认启用\`Antd Button\`，该属性已遗弃，未来版本将移除该属性|never|-|-|
|compat |测试嵌套|boolean|-|-|

"
`;

exports[`DocumentIntersection 文档模型解析[toTypeString]: 1`] = `
"{ a: string; b?: number; c: { c1: string; c2: () => string; }; }&CompatButtonProps<N>

CompatButtonProps<N>"
`;
